src/cli/commands/files/add.js

```diff
@@ -94,26 +107,62 @@ module.exports = {
       }
     }
 
-    createAddStream((err, addStream) => {
-      if (err) {
+    // if there are piped arguments, input is data to publish instead of a file
+    // path
+    if(hasPipedArgs) {
+      const data = argv.file
```

Comment:
If I do `cat someFile.txt | ipfs add` does pipe-args buffer the whole file first? Well, then it is missing the point of pipes, right?